### PR TITLE
Simplify contract release process and verify release commands and docs

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

_A few sentences describing the overall effects and goals of the pull request's commits.
What is the current behavior, and what is the updated/expected behavior with this PR?_

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: packages/docs/community/release-process/smart-contracts.md [hunk 1]
```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
...
 
-## Build and Release Process
+## Release Process
 
-Using the following scripts, new contracts can be built and deployed alongside a corresponding on-chain governance proposal. Run the following scripts from the branch with the new release contracts:
+Using the following procedure, a new contract release can be built, deployed, and proposed for upgrade automatically.
 
+### Verify Current Deployment
+
+Compile the contracts in <code>$RELEASE_BRANCH</code> branch and verify that the deployed bytecode on <code>$NETWORK</code> matches compiled bytecode.
```
Comment (nambrot) [inline packages/docs/community/release-process/smart-contracts.md:46]: Can we reference the specific tags from above? I.e. `celo-core-contracts-v${N-1}.(baklava | alfajores | mainnet)`

Comment (nambrot) [reply packages/docs/community/release-process/smart-contracts.md:46]: And maybe just include the command to checkout and compile?

File: packages/docs/community/release-process/smart-contracts.md [hunk 2]
```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
...
+yarn verify-deployed -n $NETWORK -b $RELEASE_BRANCH
+```
 
-## Script 2 ##
-# TODO: command to pull source code from current release branch
-# generate report on backwards compatibility between RC1 and current branch with new release contracts
-yarn ts-node scripts/check-backward.ts report --exclude ".*Test|Mock.*|I[A-Z].*|.*Proxy|LinkedList|SortedLinkedList|SortedLinkedListWithMedian|MultiSig.*|ReleaseGold" -o build/rc1 -n build/contracts --output_file report.json
+### Check Backward Compatibility
+
+Compile the contracts in <code>$RELEASE_BRANCH</code> and <code>$BRANCH</code> branches and check backwards compatibility against a release, subject to the following exceptions:
```
Comment (nambrot) [inline packages/docs/community/release-process/smart-contracts.md:54]: Ditto to use the specific tags branches, i.e. `celo-core-contracts-v${N-1}.(baklava | alfajores | mainnet)` and `release/celo-core-contracts/${N}`

File: packages/docs/community/release-process/smart-contracts.md [hunk 3]
```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
...
+  1. If the STORAGE version has changed, do not perform backwards compatibility checks
+  2. If the MAJOR version has changed, check storage layout is backwards compatible, but do not check contract ABI is backwards compatible
 
-## Script 3 ##
-# deploy new contracts and generate governance proposal for upgrade
-yarn truffle exec --network development ./scripts/truffle/make-release.js --build_directory build/ --report report.json --proposal proposal.json --initialize_data example-initialize-data.json'
+For changed contracts, confirm that version number in <code>$BRANCH</code> is strictly greater than <code>$RELEASE_BRANCH</code> version number.
```
Comment (nambrot) [inline packages/docs/community/release-process/smart-contracts.md:58]: I'm not too familiar with this step, but wouldn't the script just exit 1 if something is incorrect? I.e. does someone have to manually check something here and is this more meant as explanation/context?

File: packages/docs/community/release-process/smart-contracts.md [hunk 4]
```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
...
+
+```bash
+# TODO: command to pull source code from current release branch
+yarn check-versions -a $RELEASE_BRANCH -b $BRANCH -r $REPORT
 ```
 
-These scripts do the following:
+### Deploy New Contracts
+
+Build contracts on <code>$BRANCH</code> and, using corresponding backwards compatibility report <code>$REPORT</code>, deploy changed contracts to <code>$NETWORK</code>.
```
Comment (nambrot) [inline packages/docs/community/release-process/smart-contracts.md:69]: ditto on branch names

File: packages/docs/community/release-process/smart-contracts.md [hunk 5]
```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
...
+STORAGE updates are adopted by deploying a new proxy/implementation pair. Initialize new contracts with values using <code>$INITIALIZE_DATA</code> when specified.
```
Comment (nambrot) [inline packages/docs/community/release-process/smart-contracts.md:70]: What is `$INITIALIZE_DATA`?

File: packages/docs/community/release-process/smart-contracts.md [hunk 6]
```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
...
-#### Script 1:
+```bash
+yarn make-release -r $REPORT -b $BRANCH -n $NETWORK -i $INITIALIZE_DATA -p $PROPOSAL
+```
 
-1. Compiles the contracts at <code>rc1</code> branch and confirms that the compiled bytecode matches what is currently deployed on the specified network.
+### Submit Upgrade Proposal
 
-#### Script 2:
+Submit the autogenerated upgrade proposal to the Governance contract for review by voters, outputting a unique identifier.
```
Comment (nambrot) [inline packages/docs/community/release-process/smart-contracts.md:79]: Should stakeholders also verify the current deployment from the last audited commit

File: packages/docs/community/release-process/smart-contracts.md [hunk 7]
```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
...
 ```bash
-# verify bytecode of upgrade deployment
-yarn truffle exec --network development ./scripts/truffle/verify-bytecode.js --build_artifacts build/contracts --proposal ../../proposal.json
+celocli governance:show <...> --proposalID $UPGRADE_PROPOSAL_ID --jsonFile $UPGRADE_PROPOSAL
 ```
 
-This script compiles the contracts at the current branch and confirms that the compiled bytecode matches the bytecode of contracts deployed as a part of the ugrade propsal.
+### Verify Release Candidate
+
+Verify that the <code>$UPGRADE_PROPOSAL</code> proposal performs an upgrade to contract addresses which match compiled bytecode from the release candidate <code>$BRANCH</code> branch exactly. Releases are tagged as `celo-core-contracts-v${N}.(baklava | alfajores | mainnet)`.
```
Comment (nambrot) [inline packages/docs/community/release-process/smart-contracts.md:100]: I think specifically $BRANCH in this context should be the audited commit?

Reading this now, I wonder if we should only tag that commit with `celo-core-contracts-v${N}.(baklava | alfajores | mainnet)` once the proposal actually goes through?

**Review by nambrot — COMMENTED** [review]
Great improvements, just some minor clarifications

File: packages/docs/community/release-process/smart-contracts.md (hunk 1 above)
Comment (yorhodes) [reply packages/docs/community/release-process/smart-contracts.md:46]: this does checkout and compile :)

File: packages/docs/community/release-process/smart-contracts.md (hunk 3 above)
Comment (yorhodes) [reply packages/docs/community/release-process/smart-contracts.md:58]: no one has to manually check anything
this is explaining what the command does but can be reworded

File: packages/docs/community/release-process/smart-contracts.md (hunk 5 above)
Comment (yorhodes) [reply packages/docs/community/release-process/smart-contracts.md:70]: for initializing storage incompatible contracts which are being deployed anew (ie `DowntimeSlasher`)

File: packages/docs/community/release-process/smart-contracts.md (hunk 3 above)
Comment (nambrot) [reply packages/docs/community/release-process/smart-contracts.md:58]: Ah gotcha, maybe its just me but I read it as an instruction to the reader (most things in the doc are IMO)

File: packages/docs/community/release-process/smart-contracts.md (hunk 5 above)
Comment (nambrot) [reply packages/docs/community/release-process/smart-contracts.md:70]: Where does it come from? Should we have a process for this?

**Pull request merged**